theano/gof/cmodule.py

```diff
@@ -1893,7 +1893,7 @@ def join_options(init_part):
         # or 64 bit and compile accordingly. This step is ignored for ARM
         # architectures in order to make Theano compatible with the Raspberry
         # Pi, and Raspberry Pi 2.
-        if not any(['arm' in flag for flag in cxxflags]) and platform.machine() != 'armv7l':
+        if not any(['arm' in flag for flag in cxxflags]) and platform.machine() not in ['armv7l','armv6l']:
```

Comment:
@nouiz @davikrehalt This fix fails on arm 64 bit processors. On ARMv8 (i.e., ARM64) processors platform.machine() returns 'aarch64'. Hence, the check will not fail and it will include the '-m64' in the compile list of options which in turn results in a "g++: unrecognized compile option -m64" during the compile time.